File: Leetcode/1679.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        unordered_map<int,int> unmap;
        for (const auto &i : nums) {
            ++unmap[i];
        }
        int counter = 0;
        while (!unmap.empty()) {
            auto &[key, val] = *unmap.begin();
            int diff = k - key;
            if (unmap.find(diff) == unmap.end()) {
                unmap.erase(key);
            }
            else {
                if (key == diff) {
                    counter += val / 2;
                    unmap.erase(key);
                }
                else {
                    counter += min(val, unmap[diff]);
                    unmap.erase(key);
                    unmap.erase(diff);
                }
            }
        }
        return counter;
    }
};

File: Leetcode/1679/1679.go
func maxOperations(nums []int, k int) int {
	freq := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		freq[nums[i]]++
	}
	counter := 0
	for len(freq) > 0 {
		for k1, v1 := range freq {
			k2 := k - k1
			if v2, ok := freq[k2]; ok {
				if k2 == k1 {
					counter += v1 / 2
					delete(freq, k1)
				} else {
					if v1 > v2 {
						counter += v2
					} else {
						counter += v1
					}
					delete(freq, k1)
					delete(freq, k2)
				}
			} else {
				delete(freq, k1)
			}
		}
	}
	return counter
}